packages/@react-spectrum/datepicker/docs/TimeField.mdx

```diff
@@ -0,0 +1,258 @@
+
+### Granularity
+
+The `granularity` prop allows you to control the smallest unit that is displayed by a `TimeField`. By default, times are displayed with `"minute"` granularity. More granular time values can be displayed by setting the `granularity` prop to `"second"`.
+
+```tsx example
+<TimeField
+  label="Event time"
+  granularity="second"
+  defaultValue={parseAbsoluteToLocal('2021-04-07T18:45:22Z')} />
+```
+
+## Labeling
+
+A visual label should be provided for the `TimeField` using the `label` prop. If the `TimeField` is required, the `isRequired` and `necessityIndicator` props can be used to show a required state.
+
+```tsx example
+<Flex gap="size-150" wrap>
+  <TimeField label="Event time" />
+  <TimeField label="Event time" isRequired necessityIndicator="icon" />
+  <TimeField label="Event time" isRequired necessityIndicator="label" />
+  <TimeField label="Event time" necessityIndicator="label" />
+</Flex>
+```
+
+### Accessibility
+
+If a visible label isn't specified, an `aria-label` must be provided to the `TimeField` for
+accessibility. If the field is labeled by a separate element, an `aria-labelledby` prop must be provided using
+the `id` of the labeling element instead.
+
+### Internationalization
+
+In order to internationalize a `TimeField`, a localized string should be passed to the `label` or `aria-label` prop.
+When the `necessityIndicator` prop is set to `"label"`, a localized string will be provided for `"(required)"` or `"(optional)"` automatically.
+
+## Events
+
+`TimeField` accepts an `onChange` prop which is triggered whenever the time is edited by the user. The example below uses `onChange` to update a separate element with a formatted version of the date in the user's locale and local time zone. This is done by converting the date to a native JavaScript `Date` object to pass to the formatter. The `TimeField` allows editing the time components while keeping the date fixed.
+
+```tsx example
+import {getLocalTimeZone} from '@internationalized/date';
+import {useDateFormatter} from '@adobe/react-spectrum';
+
+function Example() {
+  let [date, setDate] = React.useState(parseAbsoluteToLocal('2021-04-07T18:45:22Z'));
+  let formatter = useDateFormatter({dateStyle: 'long', timeStyle: 'long'});
+
+  return (
+    <>
+      <TimeField label="Time" value={date} onChange={setDate} />
+      <p>Selected date and time: {formatter.format(date.toDate(getLocalTimeZone()))}</p>
+    </>
+  );
+}
+```
+
+## Validation
+
+`TimeField` can display a validation state to communicate to the user whether the current value is valid or invalid. Implement your own validation logic in your app and pass either `"valid"` or `"invalid"` via the `validationState` prop. The `errorMessage` prop can be used to communicate errors to the user.
+
+This example validates that the selected time is on a 15 minute increment.
+
+```tsx example
+function Example() {
+  let [time, setTime] = React.useState(new Time(9, 15));
+
+  return (
+    <TimeField
```

Comment:
The width growth of the input field because it takes its with from the longer errorMessage is ugly. Not a bug, just don't like it. I prefer if there was text wrap or a larger default size.

Reply:
Hmm yeah should probably look at that